nixos/modules/services/misc/gitlab.nix

```
@@ -368,7 +350,7 @@ in {
       };
 
       secrets.secret = mkOption {
-        type = types.str;
+        type = types.path;
```

One thing I don't like is how this breaks everybody's config with a mediocre error message (type mismatch). Alternatives are:
- Introduce `{secret,db,...}File` options for path based passwords. I don't like this because it introduces more options.
- Introduce a type `types.changed types.str types.path` for such situations, maybe even `types.changed "<reason, and remedy>" types.str types.path`. I quite like this one.

I don't consider this a blocker for this PR though.

---

@Infinisil, in nextcloud you have the `adminPass` and `adminPassFile` options next to each other. I think that's quite an ok solution.

---

@Infinisil, in nextcloud you have the `adminPass` and `adminPassFile` options next to each other. I think that's quite an ok solution.

---

As per infinisil's suggestion
```suggestion
        type = types.changed "The secret is now stored in a file, rather than plain text, to avoid it being world-visible." types.str types.path;
```